syntax = "proto3";

package syft.core.tensor;

import "proto/core/adp/entity.proto";
import "proto/lib/numpy/array.proto";
import "proto/core/io/address.proto";
import "proto/core/common/common_object.proto";

message TensorWrappedSingleEntityPhiTensorPointer {
  syft.core.adp.Entity entity = 1;
  syft.lib.numpy.NumpyProto min_vals = 2;
  syft.lib.numpy.NumpyProto max_vals = 3;
  syft.core.io.Address location =
      4; // this is to pass the client information on
  bytes scalar_manager =
      5; // this is because the VMPSM is serialized using RecursiveSerde
  syft.core.common.UID id_at_location = 6;
  string object_type = 7;
  repeated string tags = 8;
  string description = 9;
  optional bytes public_shape = 10;
}